apps/openmw/mwworld/esmstore.cpp

```diff
@@ -142,6 +142,40 @@ void ESMStore::setUp()
     mAttributes.setUp();
     mDialogs.setUp();
     mStatics.setUp();
+
+    validate();
+}
+
+void ESMStore::validate()
+{
+    // validate NPCs for non-existing class
+    for (SharedIterator<ESM::NPC> it = mNpcs.begin(); it != mNpcs.end(); ++it)
+    {
+        ESM::NPC& npc = const_cast<ESM::NPC&>(*it); // FIXME: cast
+
+        const std::string npcFaction = npc.mFaction;
+        if (!npcFaction.empty())
+        {
+            const ESM::Faction *fact = mFactions.search(npcFaction);
+            if (fact == NULL)
+            {
+                std::cout << "NPC " << npc.mName << " has unexisting faction '" << npc.mFaction << "', ignore faction." << std::endl;
```

Comment:
s/unexisting/nonexistent/g

ignoring it

Might be useful to use the ID instead of/with the name.

Reply:
Done.